### ISSUE TITLE
CS0433	The type 'SerializableAttribute' exists in both 'Autofac, Version=4.0.1.0' and 'System.Runtime, Version=4.2.0.0'

### ISSUE THREAD

**Issue by jimmymain** [autofac/Autofac.Extensions.DependencyInjection#29]
I obtain the above error on a dotnetcore 2.0 application.
My project file is as follows:

``` xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
    <CodeAnalysisRuleSet>crypto.ruleset</CodeAnalysisRuleSet>
    <TypeScriptCompileBlocked>true</TypeScriptCompileBlocked>
    <TypeScriptToolsVersion>2.7</TypeScriptToolsVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.2.1" />
    <PackageReference Include="Documentation.Analyser" Version="1.0.6538.15147" />
    <PackageReference Include="FluentValidation.AspNetCore" Version="7.5.0" />
    <PackageReference Include="Microsoft.AspNetCore" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.AspNetCore.SignalR" Version="1.0.0-preview1-final" />
    <PackageReference Include="Microsoft.AspNetCore.SpaServices" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.Extensions.Configuration" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="2.1.0-preview1-final" />
    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="2.1.0-preview1-final" />
    <PackageReference Include="Newtonsoft.Json" Version="11.0.1" />
    <PackageReference Include="System.Reactive.Core" Version="4.0.0-preview00001" />
    <PackageReference Include="Serilog.AspNetCore" Version="2.1.1" />
    <PackageReference Include="Serilog.Enrichers.Thread" Version="3.0.0" />
    <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
    <PackageReference Include="Serilog.Sinks.File" Version="4.0.1-dev-00795" />
    <PackageReference Include="Swashbuckle.AspnetCore" Version="2.2.0" />
  </ItemGroup>

  <ItemGroup>
    <None Update="CryptoBadger.db">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </None>
  </ItemGroup>

  <ItemGroup>
    <Folder Include="logs\" />
  </ItemGroup>

</Project>
```

**Comment by tillig** [comment]
We probably need to pin a later version of Autofac as the base. This was an old problem solved by an Autofac update. To work around it while we fix the A.E.DI package you should add a direct reference to Autofac - the latest version is a good choice.

**Comment by tillig** [comment]
Note to issue followers - this question was [double-filed here and on StackOverflow](https://stackoverflow.com/q/49206572/8116).

**Comment by tillig** [comment]
Looks like the fix in core Autofac took place in [v4.2.0](https://github.com/autofac/Autofac/releases/tag/v4.2.0) so that's what we'll have to update the minimum to.

**Comment by tillig** [comment]
Released as [4.2.2](https://www.nuget.org/packages/Autofac.Extensions.DependencyInjection/4.2.2)
